src/net_processing.cpp

```diff
@@ -706,6 +714,16 @@ void Misbehaving(NodeId pnode, int howmuch)
 // blockchain -> download logic notification
 //
 
+// To prevent fingerprinting attacks, only send blocks/headers outside of the
+// active chain if they are no more than a month older (both in time, and in
+// best equivalent proof of work) than the best header chain we know about.
+static bool OrphanBlockRequestAllowed(const CBlockIndex* pindex, const Consensus::Params& consensusParams)
+{
+    return (pindexBestHeader != nullptr) &&
```

Comment:
Missing `AssertLockHeld` because of `pindexBestHeader`?

Reply:
+1